src/fides/api/service/privacy_request/request_runner_service.py

```diff
@@ -415,6 +417,9 @@ async def run_privacy_request(
             ) and can_run_checkpoint(
                 request_checkpoint=CurrentStep.erasure, from_checkpoint=resume_step
             ):
+                _verify_masking_secrets(policy, privacy_request_id, resume_step)
```

Comment:
Important change! We don't cache the masking secrets until we get to the erasure step. In the event that we resume the erasure, we will error the request if masking secrets are needed for the given policy but have expired. We won't re-generate masking secrets because they need to be consistent for each erasure request.

Reply:
I think there's an edge case here where the failed step expired from the cache - we added awhile back if that was the case to let you re-process from the beginning - so resume_step will be None, but it's possible some data was already masked?